src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -158,12 +175,121 @@ export const TradeInput = ({
+            if (!accountType) throw new Error('No account type found')
+
+            const sellAssetChainAdapter = assertGetUtxoChainAdapter(chainId)
+
+            const xpub = (
+              await sellAssetChainAdapter.getPublicKey(wallet, sellAccountNumber, accountType)
+            ).xpub
+
+            const account = await sellAssetChainAdapter.getAccount(xpub)
+
+            if (!account.chainSpecific.addresses) throw new Error('No addresses found')
+
+            const addressWithEnoughBalance = account.chainSpecific.addresses.find(address => {
+              return bnOrZero(address.balance).gte(sellAmountCryptoBaseUnit)
+            })
+
+            if (!addressWithEnoughBalance) {
+              const nextReceiveAddress = await sellAssetChainAdapter.getAddress({
+                wallet,
+                accountNumber: bip44Params.accountNumber,
+                accountType,
+              })
+
+              return nextReceiveAddress
+            }
+
+            return addressWithEnoughBalance?.pubkey
+          }
+        : skipToken,
+  })
+
+  const getHasEnoughBalanceForTxPlusFees = useCallback(
+    ({
+      balanceCryptoBaseUnit,
+      amountCryptoPrecision,
+      txFeeCryptoBaseUnit,
+      precision,
+    }: {
+      balanceCryptoBaseUnit: string
+      amountCryptoPrecision: string
+      txFeeCryptoBaseUnit: string
+      precision: number
+    }) => {
+      const balanceCryptoBaseUnitBn = bnOrZero(balanceCryptoBaseUnit)
+      if (balanceCryptoBaseUnitBn.isZero()) return false
+
+      return bnOrZero(amountCryptoPrecision)
+        .plus(fromBaseUnit(txFeeCryptoBaseUnit, precision ?? 0))
+        .lte(fromBaseUnit(balanceCryptoBaseUnitBn, precision))
+    },
+    [],
+  )
+
+  const isSweepNeededArgs = useMemo(
+    () => ({
+      assetId: sellAsset.assetId,
+      address: fromAddress,
+      amountCryptoBaseUnit: sellAmountCryptoPrecision,
+      txFeeCryptoBaseUnit: tradeQuoteStep?.feeData?.networkFeeCryptoBaseUnit,
+      // Don't fetch sweep needed if there isn't enough balance for the tx + fees, since adding in a sweep Tx would obviously fail too
+      enabled: Boolean(
+        fromAddress &&
+          bnOrZero(sellAmountCryptoPrecision).gt(0) &&
+          tradeQuoteStep?.feeData?.networkFeeCryptoBaseUnit &&
+          getHasEnoughBalanceForTxPlusFees({
+            precision: sellAsset.precision,
+            balanceCryptoBaseUnit,
+            amountCryptoPrecision: sellAmountCryptoPrecision,
+            txFeeCryptoBaseUnit: tradeQuoteStep?.feeData?.networkFeeCryptoBaseUnit,
+          }),
+      ),
+    }),
+    [
+      balanceCryptoBaseUnit,
+      fromAddress,
+      getHasEnoughBalanceForTxPlusFees,
+      sellAmountCryptoPrecision,
+      sellAsset.precision,
+      tradeQuoteStep?.feeData?.networkFeeCryptoBaseUnit,
+      sellAsset.assetId,
+    ],
+  )
+
+  const { data: isSweepNeeded, isLoading: isSweepNeededLoading } =
+    useIsSweepNeededQuery(isSweepNeededArgs)
+
```

Comment:
q: do we need `isSweepNeeded` here (pun intended)?
Say there is enough of value in one address UTXO(s?), are we actually using said address without sweeping? For the sake of KISS, do we want to always sweep for Relay not to pollute `<TradeInput />` with it?

Reply:
I reverted this thing in [8aad17b](https://github.com/shapeshift/web/pull/9222/commits/8aad17b5bc9be5a528d329651d22fa59044b852e)

Now we are not even sweeping but using multiple UTXOs if needed

And see the updated context in the description of https://github.com/shapeshift/web/pull/9222#issue-2963670104